### DISCUSSION TITLE
Grove端子の電源電圧について

### DISCUSSION THREAD

**Discussion by TakSan0** [algyan/XIAOGYAN#38] [General]
Grove 端子の VDD ですが、現状3.3V 固定されていますが、選択できるようにしませんか？
世の中には 5V になっているものと 3.3V になっているものが混在し、ちょっとややこしいことになっているのは周知の事かと思います。
手持ちのデバイスが M5Stack社の製品が多いこともあり、5V多めですがちゃんと使えるのかと疑問におもって回路図等を探し回ることになり、面倒くささを感じました。

5V のデバイスはおそらく LDO がはいっていて内部に 3.3V に落としているだろうし、 i2c なので信号ラインはオープンドレインで 3.3V に釣り上げてるとおもうので、大丈夫だとはおもいます。
デバイス側に入っている LDO に 3.3V を入れても大丈夫なのか？という疑問もちょっとありますが。

いずれにせよ、回路の読めない人にとっては「よくわからないけど電圧違うので使えないのか」とあきらめる方もいるかと思います。ジャンパー一つで使用可能デバイスの選択肢が広がるならいいのかと思います。

半田ジャンパー一つだと追加部品もいらないだろうし、頻繁に色々なデバイスを付け替えたい人向けには 3pinのピンヘッダを立てればいいので、半田ジャンパーとスルーホールの２パターンを並列にするといいかと思います。

その場合電流設計がややこしくなるとか、ユニバーサルエリアの電源をどう扱うのか議論がでて、検討しないといけないかもしれませんが、如何でしょうか？

**Comment by matsujirushi** [comment]
@TakSan0 
アドバイスありがとうございます！

5Vで駆動すると信号線（Pin1, Pin2）に5V加わるGroveモジュール製品があるため、3.3V固定にしています。
（Seeed Studio XIAO ESP32C3は5Vトレラントではないため、故障のリスクあり。）

M5Stack社のGrove互換モジュールがどういうものが多いのかは知らないので、（故障リスクよりも）容易に5V供給できたほうがいいか判断つかないです。

どなたかご意見いただけませんか？

**Reply by TakSan0** [reply]
> 5Vで駆動すると信号線（Pin1, Pin2）に5V加わるGroveモジュール製品があるため、3.3V固定にしています。

なるほど、それは確かに怖いですね。
M5Stack社のは全部とはいえないのですが、みたやつは 信号ライン3.3V で電源 5V で内部で LDO で落としていました。

色々なパターンが混在していて、ややこしい仕様になってしまいましたね Groveコネクタ。
形さえ合えばつながると安易な判断で、気にしないでつないでしまう人もいるので怖いですよね。

**Reply by y-ohe** [reply]
今回の基板におけるGrove電源端子は、マイコンがXIAOシリーズなので、**3.3V固定**で良いと思います。
Groveモジュールもいろいろ展開されて、電源電圧だけでなく、信号線のレベルも異なる多数の種類にバリエーションが増えすぎていて、今では全部に対応するとこはほぼ無理になっていると思います。
とりあえず、電源線圧も信号電圧も全て3.3VのGroveモジュールにだけ対応しておけば、他のバリエーションのモジュールを間違えて繋いでしまっても、大抵の場合は動かない(使えない)だけで、**破壊することはない**と思われるからです。

**Comment by matsujirushi** [comment]
https://note.com/kuribata/n/n0c46035cc5e0

* M5Stack本体はプルアップなし。
* 純正Unit(Pb.HUB)は5Vにプルアップ（！）

お、おぅ...。なるほど...。

このWeb情報みた感じだと、I/Oに直接47を入れておかないと不安。

**Reply by matsujirushi** [reply]
ES2基板まではノイズ対策のために用意していたが、不要と判断してES3基板で除いている。

### ES2基板の一部
![image](https://user-images.githubusercontent.com/8079404/226161002-6ecb63f0-a5f9-4ec8-b96d-c4a5cdaace37.png)

**Comment by TakSan0** [comment]
> このWeb情報みた感じだと、I/Oに直接47を入れておかないと不安。

この記事見ていると、かなり不安になります。 M5Stack社のデバイスですら統一性が無い⁉

**Comment by cubicrootk** [comment]
ATmega328Pを搭載したArduino Uno R3登場したのが2011年と12年前なのですね。これは５V（16Mhz）でした。3.3V(８Mhz)でも動く互換品が出ていた気がします。Seeed社のGrove - Qwiic Hub (SKU 103020292) はGroveが5/3,3Vのどちらでも、QWIIC側は3,3Vに電源も信号レベルも変換します。これのGrove-5VからGrove-3.3V版が欲しいところです。

**Reply by matsujirushi** [reply]
XIAOGYANに5V品Groveを繋ぐ変換基板が欲しいということでしょうか？
XIAOGYAN側のGroveが3.3Vなので、SKU103020292の回路は使えないと思うのですが。🤔

**Reply by cubicrootk** [reply]
受け側のGrove基板に3端子回路等があって、電源が5/3.3V対応であるケースが多そう。古めの親側で５V信号レベルで出力する場合には、Groveケーブルを切断してレベルシフタ基板を間に入れればよいと思いますし、このユースケースは減ってきていると思います。
XIAOGYANではなくて、受け側(センサーノード)のGrove基板をM5Stackで使いまわすユースケースで発言してました。
因みにSKU103020292の回路図を見ると3.3V電源の親機(=XIAOGYAN)に使えないと感じたのですが、実際は3.3Vの電源をパススルーして使えました！　「使ってみたよ＃50」にその様子を掲載しました。
